olive/passes/onnx/append_pre_post_processing_ops.py

```diff
@@ -4,18 +4,42 @@
 # --------------------------------------------------------------------------
 import tempfile
 from pathlib import Path
-from typing import Any, Callable, Dict, List
+from typing import Any, Callable, Dict, List, Union
 
 import onnx
 from onnxruntime import __version__ as OrtVersion
+from pydantic import Field, validator
 
+from olive.common.config_utils import ConfigBase
 from olive.hardware.accelerator import AcceleratorSpec
 from olive.model import ONNXModel
 from olive.passes import Pass
 from olive.passes.onnx.common import get_external_data_config, model_proto_to_olive_model
+from olive.passes.onnx.pipeline import TENSOR_TYPE_MAP
 from olive.passes.pass_config import PassConfigParam
 
 
+class PrePostProcessorInput(ConfigBase):
+    name: str = Field(..., description="Name of the input.")
+    data_type: str = Field(..., description="Data type of the input.")
+    shape: List[Union[str, int]] = Field(..., description="Shape of the input.")
+
+    @validator("data_type", pre=True)
```

Comment:
Nitpick: I saw you have a PR to upgrade pydantic to >2.0.0. Please remember to update this.

Reply:
yes